lms/djangoapps/edraak_university/forms.py

```diff
@@ -35,15 +37,25 @@ class UniversityIDForm(forms.ModelForm):
         },
     )
 
-    section_number = forms.CharField(
+    cohort = forms.ChoiceField(
         label=_('Section Number *'),
         # TODO: Make ID format instruction course-variant, so coordinators can define it for each course.
-        help_text=_('Enter the number/name of the section e.g. 1, 2 or A, B, C depending on your university structure.'),
+        help_text=_('Select the cohort you are enrolled in.'),
     )
 
+    def __init__(self,course_pk=None, is_disabled=False, *args, **kwargs):
+        super(UniversityIDForm,self).__init__(*args, **kwargs)
+        if is_disabled:
+            for field in self.fields.keys():
+                self.fields[field].widget.attrs['readonly'] = True
+            self.fields['cohort'].widget.attrs['disabled'] = True
+        if course_pk:
+            cohorts_choices = CourseUserGroup.objects.filter(course_id=course_pk)
+            self.fields['cohort'].choices = [('','----')] + list(cohorts_choices.values_list('id','name'))
+
```

Comment:
get_cohort_names or get_course_cohorts both used a course object so I preferred passing a coure_id instead of a course object.